client/src/components/ObjectViewer/ReadOnlyValue.tsx

```diff
@@ -9,6 +10,105 @@ function shouldDiffLineForString(text: string) {
   return text?.includes('\n') || text?.includes(' ');
 }
 
+type ReadonlyValueContentProps = {
+  className: string;
+  previewMode: boolean;
+  isError: boolean;
+  truncateText: number;
+  icon: React.ReactNode;
+  text: string;
+};
+
+function ReadonlyValueContent(props: ReadonlyValueContentProps) {
+  const { className, previewMode, isError, truncateText, icon, text } = props;
+
+  const isSVG = useMemo(() => {
+    if (!text) return false;
+    const trimmedText = text.trim();
+    return trimmedText.startsWith('<svg') && trimmedText.endsWith('</svg>');
+  }, [text]);
+
+  const sanitizedSVG = useMemo(() => {
+    if (!isSVG) return null;
+
+    // Add viewBox if not present to support width adjustments for layout
+    let svgText = text;
+    if (!text.includes('viewBox=')) {
+      const widthMatch = text.match(/width="([^"]+)"/);
+      const heightMatch = text.match(/height="([^"]+)"/);
+      if (widthMatch && heightMatch) {
+        const width = widthMatch[1];
+        const height = heightMatch[1];
+        svgText = text.replace(/<svg/, `<svg viewBox="0 0 ${width} ${height}"`);
+      }
+    }
+
+    return DOMPurify.sanitize(svgText, {
+      USE_PROFILES: { svg: true, svgFilters: true },
+      ADD_TAGS: ['use'],
+      ADD_ATTR: ['xlink:href'],
+    });
+  }, [text, isSVG]);
+
+  if (isSVG && sanitizedSVG) {
+    return (
```

Comment:
Can you extract that component ? and include the svg sanitation so we can reuse elsewhere ?